src/lib.rs

```diff
@@ -101,6 +107,130 @@ impl RpcClient for ULinkZenoh {
     }
 }
 
+#[async_trait]
+impl RpcServer for ULinkZenoh {
+    async fn register_rpc_listener(
+        &self,
+        method: UUri,
+        listener: Box<dyn Fn(Result<UMessage, UStatus>) + Send + Sync + 'static>,
+    ) -> Result<String, UStatus> {
+        // Do the validation
+        if UriValidator::validate(&method).is_err() {
```

Comment:
I think according to the spec we'd want to run [`UriValidator::is_resolved()`](https://github.com/eclipse-uprotocol/uprotocol-rust/blob/493a2742b22f945184ff46f09d6992835dfad30d/src/uri/validator/urivalidator.rs#L122) since that'd check for both satisfying micro and long form (both IDs and names).

Reference to spec for UURIs [here](https://github.com/eclipse-uprotocol/uprotocol-rust/blob/493a2742b22f945184ff46f09d6992835dfad30d/src/uri/validator/urivalidator.rs#L122).

Reply:
I think same applies to the validation done in `UTransport::send()`

Reply:
I think the same applies to the validation done in `RpcClient::invoke_method()`.

Reply:
I see. So according to the spec, we should not only validate the UURI but also need to run `is_resolved()`.
But I have a question about this: since `is_resolved()` can only pass while both micro and long form are satisfied. Does that mean users should create ID (short form) everytime while using UURI even they only want to use long form (Name)?

Reply:
That's my reading of the spec, but it'd be good to have confirmation over on the issue I opened.

IMHO the other alternatives are
1. to modify the spec such that it no longer requires it or 
2. state that the `RpcClient` should also have the responsibility of calling [`uDiscovery::ResolveUri()`](https://github.com/eclipse-uprotocol/uprotocol-core-api/blob/main/src/main/proto/core/udiscovery/v3/udiscovery.proto#L127) and to fail otherwise with an appropriate `UStatus`.

Perhaps you could post over there too to get it a little more attention. What do you think?

Reply:
Agree. I've posted [here](https://github.com/eclipse-uprotocol/uprotocol-spec/issues/42#issuecomment-1882273981)

Reply:
Wanted to clarify tho, that I don't think that my point here should block this PR necessarily.

If getting follow-up on the spec takes some time, I'm in favor of marking a `TODO` in the code reflecting the open issue, merging, and moving on for now.

Reply:
Thank you for the clarification. I'll add some tests and clean up. After that, PR can be merged.
I'll add TODO to reflect the issue above.

Reply:
According to the [comment](https://github.com/eclipse-uprotocol/uprotocol-spec/issues/42#issuecomment-1887168401), we don't need to check `is_resolved()`. If there is any change in the spec, I can add it back.